pkgs/tools/misc/powerline-rs/default.nix

```diff
@@ -1,28 +1,25 @@
-{ stdenv, lib, rustPlatform, fetchFromGitLab, pkgconfig, file, perl, curl, cmake, openssl, libssh2, libgit2, libzip, Security }:
-rustPlatform.buildRustPackage rec {
+{ stdenv, Security, callPackage, lib, fetchFromGitLab }:
+
+let
   pname = "powerline-rs";
-  version = "0.1.9";
+  version = "0.2.0";
+
+  package = (callPackage ./Cargo.nix {}).rootCrate.build;
+in package.overrideAttrs (attrs: {
+  name = "${pname}-${version}";
 
   src = fetchFromGitLab {
     owner = "jD91mZM2";
-    repo = "powerline-rs";
-    #rev = version;
-
-    # Support for $COMPLETION_OUT:
-    rev = "44679385a95dd9f3ebd9b093f9ef8925610e9a23";
+    repo = pname;
+    rev = version;
 
-    sha256 = "1mxkw6ydnqjyplbki2j9pbnlhxmkw9qqw54443a3cjmn2g08jyzp";
+    sha256 = "0rqlxxl58dpfvm2idhi0vzinraf4bgiapmawiih9wxs599fnhm3y";
   };
 
-  cargoSha256 = "1d0f1c1vp1r9r3ic921xkcr59f4a45y2xbxm4gl6grhb9z6p5k7l";
```

Comment:
I feel like having a standard cargoSha256 is preferable to having a 1600+ line generated-file

Reply:
Understandable, I was thinking of the compilation speed

Reply:
generally it only needs to be built once per bump